src/connector/src/parser/json_parser.rs

```diff
@@ -102,7 +115,7 @@ impl JsonParser {
                 let accessor: UpsertChangeEvent<JsonAccess<'_, '_>, JsonAccess<'_, '_>> =
                     UpsertChangeEvent::default().with_value(JsonAccess::new(value));
 
-                match apply_row_operation_on_stream_chunk_writer(accessor, &mut writer) {
+                match apply_upsert_on_stream_chunk_writer(accessor, &mut writer) {
                     Ok(this_guard) => guard = Some(this_guard),
                     Err(err) => errors.push(err),
                 }
```

Comment:
![image](https://github.com/risingwavelabs/risingwave/assets/40830455/88a25df7-9f91-4841-ae8f-b526da5e57b7)
Seems that here still about 3% overhead. Can we use `JsonAccess` directly here? c.c. @tabVersion @wugouzi

Reply:
It seems that can be fixed by the method in https://github.com/risingwavelabs/risingwave/pull/10969. Also it will be conflict with this PR? @wugouzi

Reply:
Yes. Will change the pr accordingly.

Reply:
>  Seems that here still about 3% overhead. Can we use JsonAccess directly here?

Follow-up with it separately. Merge this in first.

Reply:
For plain format, there is no key and there is no need to build corresponding `upsertchangeeevent`. Therefore we could use payload accessor directly to write. I guess this is the point.

Reply:
I see, I just spotted it. 
```
            let mut accessor = UpsertChangeEvent::default().with_key(JsonAccess::new(key_decoded));
            if let Some(value) = value_decoded {
                accessor = accessor.with_value(JsonAccess::new(value));
            }
```

I will wait for your branch to merge.